Mathlib/Algebra/Module/Submodule/Pointwise.lean

```diff
@@ -281,4 +282,115 @@ scoped[Pointwise] attribute [instance] Submodule.pointwiseMulActionWithZero
 
 end
 
+section
+
+variable [Semiring R] [AddCommMonoid M] [Module R M]
+
+/--
+let `S ⊆ R` be a set and `N ≤ M` be a submodule, then `S • N` is the smallest submodule containing
+all `r • n` where `r ∈ S` and `n ∈ N`.
+-/
+protected def pointwiseSetSMulSubmodule : SMul (Set R) (Submodule R M) where
+  smul s N := sInf { p | ∀ ⦃r : R⦄ ⦃n : M⦄, r ∈ s → n ∈ N → r • n ∈ p }
```

Comment:
Is this is the same as the following?
```suggestion
  smul s N := ⨅ r, r • N
```

Reply:
Or perhaps
```suggestion
  smul s N := sInf <| (. • N) '' s
```
If you don't like the bInf

Reply:
They are the same, but I think defining it to be `sInf ...` is very intuitive, it is the smallest submodule containing all the `r \bu n` for `r \in s` and `n \in N`

Reply:
Instead of changing definition, maybe I just prove that `sInf { p | ∀ ⦃r : R⦄ ⦃n : M⦄, r ∈ s → n ∈ N → r • n ∈ p }` is ⨆ r ∈ s, r • N?

Reply:
Correct me if I am wrong here:
`Submodule.pointwiseDistribMulAction` wants `[SMulCommClass α R M]`, so when we write down `r • N`, we need a type class `SMulCommClass R R M`, but `R` is assumed to be semiring here not necessarily commutative, so we can't write `r • N`?

Reply:
So I proved this
```lean
lemma set_smul_submodule_eq_iSup [SMulCommClass R R M] :
    s • N = ⨆ (r ∈ s), r • N := ...
```

If we define it as `sInf`, then we don't need `SMulCommClass R R M` assumption